openhtf/io/output/mfg_inspector.py

```diff
@@ -381,11 +382,14 @@ def locked_get(self):
     def locked_put(self, credentials):
       self._credentials = credentials
   # pylint: enable=invalid-name,missing-docstring
-
-  def __init__(self, user, keydata, token_uri=TOKEN_URI):
+  
+  def __init__(self, user, keydata, token_uri=TOKEN_URI,
+               error_message=None, filename_pattern_fail=None):
```

Comment:
What would the use-case be for customizing the error message versus simply logging something like 'Upload failed; falling back to writing file to local filesystem' or the like?